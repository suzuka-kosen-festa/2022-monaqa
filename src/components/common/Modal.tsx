import React, { useState } from 'react'
import type { FC } from 'react'
import tw from 'twin.macro'

import { Button } from './Button'
import { Text } from './Text'

interface ModalInterface {
  children: React.ReactNode
  name: string
  buttonText1: string
  buttonText2: string
  onClick: () => void
}

const ModalArea = tw.div`fixed inset-0 z-10 overflow-y-auto bg-gray-700 bg-opacity-70 `
const ModalBox = tw.div`flex items-center min-h-screen px-4 py-8`
const DiaLog = tw.div`relative w-full max-w-lg p-4 mx-auto bg-white rounded-md shadow-lg text-center space-y-4 space-x-4`

const Modal: FC<ModalInterface> = ({
  children,
  name,
  buttonText1,
  buttonText2,
  onClick,
}) => {
  const [showModal, setShowModal] = useState(false)
  /* Todo: create portalやる const root = document.getElementById('modal') */
  return (
    <>
      <Button buttonKey="accept" onClick={() => setShowModal(true)}>
        {children}
      </Button>
      {showModal ? (
        <ModalArea>
          <ModalBox>
            <DiaLog>
              <Text>{name}さんの入場を記録しますか？</Text>
              <Button buttonKey="reject" onClick={() => setShowModal(false)}>
                {`${buttonText1}`}
              </Button>
              <Button buttonKey="accept" onClick={onClick}>
                {`${buttonText2}`}
              </Button>
            </DiaLog>
          </ModalBox>
        </ModalArea>
      ) : null}
    </>
  )
}

export { Modal }
