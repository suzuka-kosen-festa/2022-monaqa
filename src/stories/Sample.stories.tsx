import type { ComponentStoryObj, ComponentMeta } from '@storybook/react'
import { Sample } from '../components/Sample'

type T = typeof Sample
type Story = ComponentStoryObj<T>

export default {
  component: Sample,
} as ComponentMeta<T>

export const Default: Story = {}
