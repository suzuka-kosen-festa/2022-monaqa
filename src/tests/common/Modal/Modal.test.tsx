import { composeStories } from '@storybook/testing-react'
import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import React from 'react'
import * as stories from '../../../stories/common/Modal.stories'

const { Default } = composeStories(stories)

describe('(components/common) Modal', () => {
  test('take snap shot', () => {
    const { container } = render(<Default />)
    expect(container).toMatchSnapshot()
  })
})
