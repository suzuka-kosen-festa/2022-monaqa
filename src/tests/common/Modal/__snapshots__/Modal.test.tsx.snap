// Vitest Snapshot v1

exports[`(components/common) Modal > take snap shot 1`] = `
<div>
  <button
    class="css-177jtcm-Button"
  >
    <p
      class="css-7rlnkq-Button"
    >
      This is test
    </p>
  </button>
</div>
`;
